### ISSUE TITLE
Vidonn X6 Support and Lua

### ISSUE THREAD

**Issue by romcok** [dgomes/openvidonn#4]
Hi Diogo,
this is great project and I would like to ask you to add the support to X6 band. It would be great also if you added support for Lua - providing us with the option to make our own functions. I would love to help you with this, but I am not very good at Java.

**Comment by dgomes** [comment]
Hi,

I don't have the X6 band, so unfortunately I'm not able to support it.

I also don't know how Lua support would workout in an android app... OpenVidonn simply reads the data that is already available in the bracelet, It doesn't do any calculations.

**Comment by javaskater** [comment]
I just bought the X6 that is why I started to follow the dgomes GitHub repository.
I first want to thank him for his work ..
I am a JAVAEE + Web Developper I am now starting (privately) with Android programming. So as soon as I have time left I will start forking dgomes repo (if dgomes agrrees) while trying to write an X6 version of his app ...

**Comment by dgomes** [comment]
Hi javaskater, 

You more than welcome to fork and contribute :)

I'll take any pull requests I receive from the community

**Comment by romcok** [comment]
Can you tell me how to pull commands from the bracelet? I want to do some tests.

@javaskater +1 
Support for the X6 would help me, official application crashes every time I select an item from the menu.

**Comment by dgomes** [comment]
I used https://itunes.apple.com/us/app/lightblue/id639944780?mt=12

**Comment by computerlyrik** [comment]
@romcok: I am in - on X6 support.

Need any help?

Seems like there are fewer characteristics. Anyone knows a good BLE listener (working within linux)

**Comment by romcok** [comment]
@dgomes nice thanks

I put to work an older version of the official application, so when I have more time I will look at it.

**Comment by computerlyrik** [comment]
What i found out:

```
    public static String VIDONN_SERVICE = "0000ffe0-0000-1000-8000-00805f9b34fb";
    public static String VIDONN_OPERATION_READ = "0000ffe1-0000-1000-8000-00805f9b34fb";
    public static String VIDONN_OPERATION_NOTIFICATION = "0000ffe2-0000-1000-8000-00805f9b34fb";
    public static String VIDONN_OPERATION_WRITE = "0000ffe4-0000-1000-8000-00805f9b34fb";

    public static String VIDONN_SERVICE2 = "0000ffe5-0000-1000-8000-00805f9b34fb";
    public static String VIDONN_OPERATION_READ_CURRENT = "0000ffe9-0000-1000-8000-00805f9b34fb";
```

But these are not enough characteristics to read all the possible measurements.
So digging in Source gives something like:

```
OP_X6_R_AlarmClock = 0x69
OP_X6_R_CurrentValue = 0x3
OP_X6_R_HistoryDetail = 0x66
OP_X6_R_HistoryMap = 0x65
...
```

Anyone knows how this fits into BLE-Protocol?

**Comment by computerlyrik** [comment]
I am having a rudimentary setup here with the original code running and debugger working. But i have no idea how to read correct values.

I do get some byte[6] array with some integers from VIDONN_OPERATION_READ - but always the same and no match to display data on Vidonn X6

**Comment by computerlyrik** [comment]
`VIDONN_OPERATION_READ_CURRENT` gives an Array with [0]=Steps as displayed on wristband; other three fields are 0

**Comment by dgomes** [comment]
Forget about that, you don’t know their encodings

I would use light blue and check all the characteristics that are available, it’s easy to browse. Then would figure out what matches what.

**Comment by computerlyrik** [comment]
@dgomes sure...but i do not have a mac. Is there any way or similar SW to get run on linux? (or better - on android directly?)

PS: reading `VIDONN_OPERATION_READ` after communication with original app changes - and stays the same...i think it is some kind of stateful protocol (??)

**Comment by thelemmings** [comment]
Hello, I am working on a Windows client for the X6 and haven't been able to figure out the protocol either. Have you made any advances on your research ?

PS: I was also able to get the current steps, although it doesn't seem to be always a bit behind what the bracelet displays.

**Comment by sriraman** [comment]
Do anyone found a way to use X6?

**Comment by computerlyrik** [comment]
Not here.

Using Xiaomi Mi Fit extensively.

Want the X6?

**Comment by sriraman** [comment]
Yes. Now, I'm trying to build it for X6. So, If you give any reference, It will be useful for me. 
Can you share the Mi fit link?

Thanks,

**Comment by hipio** [comment]
I assume that nobody got this working for X6, is that correct?

**Comment by cosmojg** [comment]
@javaskater @sriraman @thelemmings @romcok Did you guys have any success in making something that works for X6/X6S? I'm definitely gonna try so any tips would be great! :)
